src/scroll-timeline-css-parser.js

```diff
@@ -159,10 +161,28 @@ export class StyleParser {
     return null;
   }
 
+  getAnonymousViewTimelineOptions(timelineName, target) {
+    const options = this.anonymousViewTimelineOptions.get(timelineName);
+    if(options) {
+      return {
+        subject: target,
+        axis: (options.axis ? options.axis : 'block'),
+        inset: (options.inset ? options.inset : 'auto'),
+      };
+    }
+
+    return null;
+  }
+
   getViewTimelineOptions(timelineName, target) {
+    const anonymousTimelineOptions = this.getAnonymousViewTimelineOptions(timelineName, target);
+    if(anonymousTimelineOptions)
+      return anonymousTimelineOptions;
+
     for (let i = this.subjectSelectorToViewTimeline.length - 1; i >= 0; i--) {
       const options = this.subjectSelectorToViewTimeline[i];
       if(options.name == timelineName) {
+        // TODO: The subject is the target, no? If yes, then can remove this extra check.
```

Comment:
This block appears to be for resolving something of the form:
animation-timeline: --name;
view_timeline: --name; // In ancestor element.

In this case I believe that target is the target element for the animation, which is not necessarily the subject.



Also findPreviousSibling....Selector will need some refactoring for supporting timeline attachments as the current is using stale scoping rules.  Please add a TODO.

Reply:
Pushed new commits to resolve this.